x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -405,12 +425,10 @@ public void onResponse(SearchResponse response) {
 
         @Override
         public void onFailure(Exception exc) {
-            if (searchResponse.get() == null) {
-                // if the failure occurred before calling onListShards
-                searchResponse.compareAndSet(null,
-                    new MutableSearchResponse(-1, -1, null, threadPool.getThreadContext()));
-            }
-            searchResponse.get().updateWithFailure(exc);
+            // if the failure occurred before calling onListShards
+            searchResponse.compareAndSet(null, new MutableSearchResponse(-1, -1, null, threadPool.getThreadContext()));
+            searchResponse.get().updateWithFailure(new ElasticsearchStatusException("error while executing search",
```

Comment:
is the additional wrapping ok? I think it's odd that we have to have it, but useful to clarify where errors come from: async search or search execution.

Reply:
++